test/conformance/ingress/headers.go

```diff
@@ -25,9 +25,114 @@ import (
 	"k8s.io/apimachinery/pkg/util/sets"
 	"knative.dev/networking/pkg/apis/networking"
 	"knative.dev/networking/pkg/apis/networking/v1alpha1"
+	"knative.dev/pkg/ptr"
+	"knative.dev/serving/pkg/network"
 	"knative.dev/serving/test"
 )
 
+// TestTagHeaders verifies that an Ingress properly dispaches to backends based on the tag header
+//
+// See proposal doc for reference:
+// https://docs.google.com/document/d/12t_3NE4EqvW_l0hfVlQcAGKkwkAM56tTn2wN_JtHbSQ/edit?usp=sharing
+func TestTagHeaders(t *testing.T) {
+	t.Parallel()
+	clients := test.Setup(t)
+
+	name, port, cancel := CreateRuntimeService(t, clients, networking.ServicePortNameHTTP1)
+	defer cancel()
```

Comment:
we're switching to new `t.Cleanup` method.
Here it'll be `t.Cleanup(cancel)`. Same below.

Reply:
Nice, didn't know they added `t.Cleanup`. We can make all the helpers just clean up after themselves (separate story).